mir_eval/io.py

```diff
@@ -477,3 +477,86 @@ def load_key(filename, delimiter=r'\s+'):
         warnings.warn(error.args[0])
 
     return key_string
+
+
+def load_ragged_time_series(filename, dtype=float, delimiter=r'\s+',
+                            header=False):
+    r"""Utility function for loading in data from a delimited time series
+    annotation file with a variable number of columns.
+    Assumes that column 0 contains time stamps and columns 1 through n contain
+    values. n may be variable from time stamp to time stamp.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the annotation file
+    dtype : function
+        Data type to apply to values columns.
+    delimiter : str
+        Separator regular expression.
+        By default, lines will be split by any amount of whitespace.
+    header : bool
+        Indicates whether a header row is present or not.
+        By default, assumes no header is present.
+
+    Returns
+    -------
+    times : np.ndarray
+        array of timestamps (float)
+    values : list of np.ndarray
+        list of arrays of corresponding values
+
+    """
+    # Initialize empty lists
+    times = []
+    values = []
+
+    # Create re object for splitting lines
+    splitter = re.compile(delimiter)
+
+    # Keep track of whether we create our own file handle
+    own_fh = False
+    # If the filename input is a string, need to open it
+    if isinstance(filename, six.string_types):
+        # Remember that we need to close it later
+        own_fh = True
+        # Open the file for reading
+        input_file = open(filename, 'r')
```

Comment:
this could leak file descriptors if there's an error anywhere below (before the call to close).

For this sort of thing, it's better to make a wrapper function that returns the file descriptor, and then use a context manager: `with my_open(buf_or_str, 'r') as fdisc: ....`

See [here](https://github.com/marl/jams/blob/master/jams/core.py#L58-L59) for an example of this in jams.